src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -119,14 +109,18 @@ public async Task InvokeAsync(HttpContext context)
                     scope.OnEvaluating += (_, __) => PopulateScope(context, scope);
                 });
 
-                var transaction = hub.CreateTransaction(transactionName, "http.server");
+                var transaction = hub.CreateTransaction(
+                    // Try to get the route template or fallback to the request path
+                    context.TryGetRouteTemplate() ?? context.Request.Path,
```

Comment:
@marandaneto @HazAT what do we do when we can't figure out a route template for a transaction name? Don't create the transaction at all?

Falling back to the actual route would be noise I assune?

@Tyrrrz or this fallback means no template was used so it's safe?

Reply:
@bruno-garcia this checks for two known way to define routes, i.e. the routing middleware and the MVC middleware (legacy). Theoretically, the user can use some other custom routing middleware and then we won't be able to extract the route. But for regular use cases, this would mean that there is no template.